mathics/builtin/base.py

```diff
@@ -70,22 +70,35 @@ def has_option(options, name, evaluation):
 
 class Builtin:
```

Comment:
Why is this not called BuiltinBase like our other Base classes?

Reply:
maybe because we want to say that ``ArcTan`` is a ``Builtin`` symbol.

Reply:
I don't understand. I might like to say ArcTan is a builtin function. Or that it is a trigonometric builtin function or many things like that. If there were common properties of built-in functions as opposed to built-in variables, then sure we could create a BuiltinFunction class which is a subclass of the Builtin (base class). 

But right now as far as I can tell, there are not properties of builtin functions that we seem to want to capture. Or trigonometric builtin functions either. And when they appear we can add them.

And `Builtin` is also used for builtin _objects_ like `DateTimeObject` and `ColorObject`. 

We work in Python where we build stuff up from. From an OO programmer's point of view, and a programmer using Mathics in particular, there are things like Base classes that you don't create instances from directly but must be subclassed. And `Builtin` is one of them. 

One of the things I personally find annoying is the looseness, vagueness (if not sometimes outright wrong) terminology backed by a lack of description of what is meant. And then that has probably lead to the object used in a way that it wasn't intended -- possibly, it is hard to tell. It has had very real tangible consequences that we even figure out how to improve the code because we can't even agree or figure what's going on. Instead it has been more like - here's a problem. Here is some code fix that addresses this. Having code that follows established conventions and principles suggests to us and others to determine when there is a problem how something should be addressed. 

Here, If there is code that instantiates a Builtin object and then starts using it (rather than subclassing it or using one of the subclasses) then that code is probably wrong.

Reply:
@rocky, OK. My unique concern is that most of the symbols are subclasses of ``Builtin``, so the change would affect almost all the builtin sub-modules. But yes, if you think it is relevant at this step, go ahead.

Reply:
@TiagoCavalcante - your thoughts?

Reply:
@rocky I'll say the truth, I don't see a problem with Builtin, it was very clear to me what it made after reading the documentation, and [for some unknown reason] it seems that renaming it to BuiltinBase would make its meaning less clear.

Reply:
Ok. Then let's keep the name as it currently is.